src/commands/deployWorkspaceProject/deploymentConfiguration/workspace/DeploymentConfigurationListStep.ts

```diff
@@ -0,0 +1,59 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { AzureWizardPromptStep, nonNullProp, type IAzureQuickPickItem, type IWizardOptions } from "@microsoft/vscode-azext-utils";
+import { localize } from "../../../../utils/localize";
+import { type DeploymentConfigurationSettings } from "../../settings/DeployWorkspaceProjectSettingsV2";
+import { dwpSettingUtilsV2 } from "../../settings/dwpSettingUtilsV2";
+import { type WorkspaceDeploymentConfigurationContext } from "./WorkspaceDeploymentConfigurationContext";
+
+export class DeploymentConfigurationListStep extends AzureWizardPromptStep<WorkspaceDeploymentConfigurationContext> {
+    public async prompt(context: WorkspaceDeploymentConfigurationContext): Promise<void> {
+        const deploymentConfigurations: DeploymentConfigurationSettings[] | undefined = await dwpSettingUtilsV2.getWorkspaceDeploymentConfigurations(nonNullProp(context, 'rootFolder'));
+        if (!deploymentConfigurations?.length) {
+            return;
+        }
+
+        context.deploymentConfigurationSettings = (await context.ui.showQuickPick(this.getPicks(deploymentConfigurations), {
+            placeHolder: localize('chooseDeployConfigurationSetting', 'Select an app configuration to deploy'),
+            suppressPersistence: true,
+        })).data;
+    }
+
+    public shouldPrompt(context: WorkspaceDeploymentConfigurationContext): boolean {
+        return !context.deploymentConfigurationSettings;
+    }
+
+    public async getSubWizard(context: WorkspaceDeploymentConfigurationContext): Promise<IWizardOptions<WorkspaceDeploymentConfigurationContext> | undefined> {
+        if (!context.deploymentConfigurationSettings) {
+            return undefined;
+        }
+
+        return {
+            executeSteps: [
+                // Todo: Shallow (local fs) validation step(s)
+                // Todo: Deep (Azure resource) validation step(s)
+            ]
+        };
+    }
+
+    private getPicks(deploymentConfigurations: DeploymentConfigurationSettings[]): IAzureQuickPickItem<DeploymentConfigurationSettings | undefined>[] {
+        const picks: IAzureQuickPickItem<DeploymentConfigurationSettings | undefined>[] = deploymentConfigurations.map(deploymentConfiguration => {
+            return {
+                label: deploymentConfiguration.label ?? localize('unnamedApp', 'Unnamed app'),
```

Comment:
If there's no label, I think we should prompt for a label as part of the wizard.

Another idea; when we convert V1 settings to V2, the label should be called something like "Converted from "$APP_NAME" settings". In this case, we would also prompt them for the label.